### PR TITLE
Fix for the test action so it actually works.

### PR DESCRIPTION
**Pull request by Frodo45127**
Changes so far:
- Fixed test compilation on push for master and develop branches.
- Moved a few more helpers from Rldx to CommonLibs.
- Moved Game ids to CommonLibs.
- Removed all uses of Qt outside QtRenderingWidget and QtTesterApp.
- Fixed a bunch of circular dependencies between ImportExport, Rldx and QtRenderingWidget.
- Standarized include mess to make sure all includes are relatives to either current file or root.
- Enabled compile-time optimizations on release mode.
- Removed a lot of unused includes.
- Temporarily disabled the anim callback, as its needs to be updated like the assets one, but it's currently unused.

Todo:
- Do a second pass removing unused includes.
- See what to do with all the files that have code but are unused.

### CODE REVIEW

File: QtTesterApp/QtTestAppView.cpp [hunk 1]
```diff
@@ -76,11 +76,6 @@ QStringList DEBUG_GetAllFiles(const QString& path, const QString& extension)
 // TODO: REMOVE and cleanup 
 void QtMainWindowView::DEBUG_InitRenderView()
 {
-	// TODO: DOESN't WORK, get set to null, as the release version is called, make better debugging code, that doesn't affect the widget, maybe
```
Comment (mr-phazer) [inline QtTesterApp/QtTestAppView.cpp:79]: This sets the debug FileGetter (stanThis sets the debug FileGetter (stanard-in for the release asset Calback) why is this deleted?

File: QtRenderingWidget/ExternFunctions/Creators.cpp [hunk 2]
```diff
@@ -2,28 +2,94 @@
...
+			newStream.Read(dest.data(), dest.size());
+		}
+	}
+	
+	// If we have callback, use it for getting the files.
+	else {
+
+		QList<QString>* filesToFetchQt = new QList<QString>();
+		QList<QByteArray>* outBinFilesQt = new QList<QByteArray>();
+
```
Comment (mr-phazer) [inline QtRenderingWidget/ExternFunctions/Creators.cpp:57]: Memory leak,should NOT use NEW here

Just declare a stack varable
```
QList<QString> filesToFetchQt;
QList<QByteArray> outBinFilesQt;
```

File: QtRenderingWidget/ExternFunctions/Creators.cpp [hunk 3]
```diff
@@ -2,28 +2,94 @@
...
+		for (const std::wstring& i : *filesToFetch) {
+			QString s = QString::fromWCharArray(i.c_str());
+			filesToFetchQt->push_back(s);
+		}
+
+		AssetFetchCallBackStored(filesToFetchQt, outBinFilesQt);
```
Comment (mr-phazer) [inline QtRenderingWidget/ExternFunctions/Creators.cpp]: So with variable above them with their addresses, and NO memory leakl
`AssetFetchCallBackStored(&filesToFetchQt, &outBinFilesQt);`

**Review by mr-phazer — CHANGES_REQUESTED** [review]

File: QtTesterApp/QtTestAppView.cpp (hunk 1 above)
Comment (Frodo45127) [reply QtTesterApp/QtTestAppView.cpp:79]: The file getter was rewritten so if the static that contains it is a nullptr, it uses the debug code. If it's set, it uses the release code.

File: QtRenderingWidget/ExternFunctions/Creators.cpp (hunk 3 above)
Comment (Frodo45127) [reply QtRenderingWidget/ExternFunctions/Creators.cpp]: Same for the one above, should be fixed in here: https://github.com/mr-phazer/QtRenderingWidget/pull/3/commits/c3fcd9e335d9dedac08b969e09f1aa1ec01db780

File: QtRenderingWidget/ExternFunctions/Creators.cpp (hunk 2 above)
Comment (Frodo45127) [reply QtRenderingWidget/ExternFunctions/Creators.cpp:57]: Should be fixed now in this commit: https://github.com/mr-phazer/QtRenderingWidget/pull/3/commits/c3fcd9e335d9dedac08b969e09f1aa1ec01db780

File: Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp [hunk 4]
```diff
@@ -147,28 +147,31 @@ namespace rldx
 		}*/
 	}
 
-	void DxResourceManager::GetResourcesFromCallBack(QList<QString>& qstrMissingFiles, QList<QByteArray>& destBinaries)
+	void DxResourceManager::GetResourcesFromCallBack(std::vector<std::wstring>* qstrMissingFiles, std::vector<std::vector<unsigned char>>* destBinaries)
```
Comment (mr-phazer) [inline Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp]: In this case is proper c++ to do
`void DGetResourcesFromCallBack(const std::vector<std::wstring>& qstrMissingFiles, std::vector<std::vector<uint8_t>>& destBinaries);`

In c++ alwayus use references if possible, and unt8_t is like the u8 i rust, it is used when you need to sure of the size

File: Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp [hunk 5]
```diff
@@ -147,28 +147,31 @@ namespace rldx
...
 	{
 		if (!sm_assetCallBack) {
 			throw std::exception("No asset callback function set");
 		}
 
-		sm_assetCallBack(&qstrMissingFiles, &destBinaries);
+		sm_assetCallBack(qstrMissingFiles, destBinaries);
```
Comment (mr-phazer) [inline Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp]: And ofc change this to
`sm_assetCallBack(qstrMissingFiles, destBinaries);`

**Review by mr-phazer — CHANGES_REQUESTED** [review]

File: Rldx/Rldx/Rendering/DXSwapChain.h [hunk 6]
```diff
@@ -69,7 +69,7 @@ namespace rldx
 		void Resize(ID3D11Device* device, ID3D11DeviceContext* deviceContext, UINT width, UINT height);
 		void Present(ID3D11DeviceContext* poDXDeviceContext);
 		DxTexture* GetBackBuffer() { return &m_BackBufferTexture; };
-		void UpdateViewPort(ID3D11DeviceContext* _pDeviceContext, HWND hwnWindowHandle = NULL);
```
Comment (mr-phazer) [inline Rldx/Rldx/Rendering/DXSwapChain.h:72]: THis paraam "HWND" was actually meant to update/resize the current window, seem I didn't finish i

File: Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp (hunk 4 above)
Comment (Frodo45127) [reply Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp]: Fixed in the last commit.

File: Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp (hunk 5 above)
Comment (Frodo45127) [reply Rldx/Rldx/Managers/ResourceManager/DxResourceManager.cpp]: Fixed in the last commit.

**Review by mr-phazer — CHANGES_REQUESTED** [review]

File: Rldx/Rldx/Rendering/DXSwapChain.h (hunk 6 above)
Comment (Frodo45127) [reply Rldx/Rldx/Rendering/DXSwapChain.h:72]: It was unused in the two versions of the function I've seen, so I think it'll be better to remove it until a use for it is found.

Comment (mr-phazer) [reply Rldx/Rldx/Rendering/DXSwapChain.h:72]: OK
